airbyte-webapp/src/core/form/schemaToFormBlock.ts

```diff
@@ -37,20 +38,49 @@ export const jsonSchemaToUiWidget = (
   }
 
   if (jsonSchema.oneOf?.length && jsonSchema.oneOf.length > 0) {
-    const conditions = Object.fromEntries(
-      jsonSchema.oneOf.map((condition) => {
-        if (typeof condition === "boolean") {
-          return [];
-        }
-        return [condition.title, jsonSchemaToUiWidget({ ...condition, type: jsonSchema.type }, key, path)];
-      })
+    let possibleConditionSelectionKeys: Set<string> | null = null as Set<string> | null;
```

Comment:
Yes, otherwise typescript tries to be smart and assumes it will always be `null`:
<img width="515" alt="Screenshot 2022-12-13 at 16 19 14" src="https://user-images.githubusercontent.com/1508364/207372573-fc9c1f6b-e047-40f3-85f7-560077c46c03.png">

However, we can omit the type of the variable. Improved.